service/worker/workerdeployment/workflow.go

```diff
@@ -115,7 +117,7 @@ func (d *WorkflowRunner) run(ctx workflow.Context) error {
 }
 
 func (d *WorkflowRunner) validateSetCurrent(args *deploymentspb.SetCurrentVersionArgs) error {
-	if d.State.CurrentVersion != args.Version {
+	if d.State.RoutingInfo.CurrentVersion != args.Version {
```

Comment:
Nit: I think it's more idiomatic in Go, and more readable, to structure the function like
```
func foo(...) error {
   if errorCondition {
      return err
   }
   return nil
}
```

Returning success at the end and errors in the middle makes it easier to read. The section "A longer example" in https://8thlight.com/insights/exploring-error-handling-patterns-in-go this article also calls this out with another example.